lang/src/org/partiql/lang/util/PluginExtensions.kt

```diff
@@ -0,0 +1,40 @@
+package org.partiql.lang.util
+
+import OTS.ITF.org.partiql.ots.Plugin
+import OTS.ITF.org.partiql.ots.type.ScalarType
+
+class TypeRegistry(types: List<ScalarType>) {
+    private var namesToType: Map<String, ScalarType> = types.flatMap { scalarType ->
+        scalarType.names.map { typeAlias -> typeAlias to scalarType }
+    }.associate { it.first to it.second }
+
+    fun getTypeByName(name: String): ScalarType? = namesToType[name]
```

Comment:
It should be a private/internal class. However, currently the fact that the public class `PartiqlPhysicalSanityValidator`'s constructor takes `TypeRegistry` as one argument requires `TypeRegistry` to be public. The 2 solutions I can come up with are: 
1. Use a `Plugin` to construct `PartiqlPhysicalSanityValidator`, and generate `TypeRegistry` from the `Plugin` once again. 
2. Make `PartiqlPhysicalSanityValidator` internal and use it to validate an AST right after parsing (instead of right before compiling). 

I personally prefer the second solution. What do you think?

Reply:
I prefer 2 as well. The validator shouldnt have been made public. Also, why are we validating the physical plan? Any validation should be done on the logical plan, and production of a physical plan from a logical plan is a minimal operation

Reply:
Also, as for whether we should throw error in case that we cannot find one entry in the type registry, I don't have strong opinion. Because in case we want to handle this case other than throwing an error, null handling or error handling seems identical to me.